### PR TITLE
Add RigidCameraController

### PR DESCRIPTION
**Pull request by zpl-zak**

### CODE REVIEW

**Comment by drummyfish** [comment]
Seems great but the new controller is mostly copy-paste from the free controller, I guess it should be its subclass - I'd add a virtual method handleMovement(Vec3 posDiff, Vec3 angleDiff) to free camera controller that would be called from within the update method, and would be overriden in the rigid controller to control the actual entity.

**Comment by zaklaus** [comment]
Yes, I'm aware of code redundancy between those controllers, considering the fact they might be useful later on, I'll make it that way, as suggested.

File: components/controllers/camera_controller.hpp [hunk 1]
```diff
@@ -13,9 +13,14 @@ class CameraController
     CameraController(MFRender::Renderer *renderer, MFInput::InputManager *inputManager);
     virtual void update(double dt)=0;
 
+    void setSpeed(double speed)         { mSpeed = speed;         };
```
Comment (drummyfish) [inline components/controllers/camera_controller.hpp]: If we consider having player/car fixed camera controllers later as subclasses to this, does this method make sense here? That's why I originally put it to the free camera.

Comment (zaklaus) [reply components/controllers/camera_controller.hpp]: Will be moved as well

**Pull request merged**